File: java-samples/Tokens/spaceships-javaAPIs/contracts/src/test/java/net/corda/samples/spaceships/contracts/ContractTests.java
package net.corda.samples.spaceships.contracts;

import com.r3.corda.lib.tokens.contracts.utilities.AmountUtilities;
import com.r3.corda.lib.tokens.money.FiatCurrency;
import net.corda.core.identity.CordaX500Name;
import net.corda.samples.spaceships.states.SpaceshipTokenType;
import net.corda.testing.core.TestIdentity;
import net.corda.testing.node.MockServices;
import org.junit.Test;
import static net.corda.testing.node.NodeTestUtils.ledger;

public class ContractTests {
    private final MockServices ledgerServices = new MockServices();
    TestIdentity Operator = new TestIdentity(new CordaX500Name("Alice",  "TestLand",  "US"));

    @Test
    public void issuerAndRecipientCannotHaveSameEmail() {
        SpaceshipTokenType tokenPass = new SpaceshipTokenType(
                Operator.getParty(),"CordaNumber1","Earth", 9, AmountUtilities.amount(1000, FiatCurrency.getInstance("USD")) ,true);
        SpaceshipTokenType tokenFail = new SpaceshipTokenType(
                Operator.getParty(),"CordaNumber1","", 9, AmountUtilities.amount(1000, FiatCurrency.getInstance("USD")) ,true);

        ledger(ledgerServices, l -> {
            l.transaction(tx -> {
                tx.output(SpaceshipTokenContract.CONTRACT_ID, tokenFail);
                tx.command(Operator.getPublicKey(), new com.r3.corda.lib.tokens.contracts.commands.Create()); // Wrong type.
                return tx.fails();
            });
            l.transaction(tx -> {
                tx.output(SpaceshipTokenContract.CONTRACT_ID, tokenPass);
                tx.command(Operator.getPublicKey(), new com.r3.corda.lib.tokens.contracts.commands.Create()); // Wrong type.
                return tx.verifies();
            });
            return null;
        });
    }
}

File: kotlin-samples/Features/cordaService-autopayroll/contracts/src/test/kotlin/net/corda/samples/autopayroll/ContractTests.kt
package net.corda.samples.autopayroll

import net.corda.core.identity.CordaX500Name
import net.corda.samples.autopayroll.contracts.MoneyStateContract
import net.corda.samples.autopayroll.states.MoneyState
import net.corda.testing.core.TestIdentity
import net.corda.testing.node.MockServices
import net.corda.testing.node.ledger
import org.junit.Test

class ContractTests {
    private val ledgerServices = MockServices()
    val partya = TestIdentity(CordaX500Name(organisation = "Alice", locality = "TestLand", country = "US"))
    var partyb = TestIdentity(CordaX500Name("Bob", "TestLand", "US"))

    @Test
    fun `No Negative PayCheck Value`() {
        val tokenPass = MoneyState(10, partyb.party)
        val tokenFail = MoneyState(-10, partyb.party)

        ledgerServices.ledger {
            transaction {
                output(MoneyStateContract.ID, tokenFail)
                command(partya.publicKey, MoneyStateContract.Commands.Pay())
                this.fails()
            }
        }
        ledgerServices.ledger {
            transaction {
                output(MoneyStateContract.ID, tokenPass)
                command(partya.publicKey, MoneyStateContract.Commands.Pay())
                this.verifies()
            }
        }
    }
}